pkg/middlewares/passtlsclientcert/pass_tls_client_cert.go

```diff
@@ -104,11 +104,77 @@ func (p *passTLSClientCert) GetTracingInformation() (string, ext.SpanKindEnum) {
 
 func (p *passTLSClientCert) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	ctx := middlewares.GetLoggerCtx(req.Context(), p.name, typeName)
+	logger := log.FromContext(ctx)
+
+	if p.pem {
+		if req.TLS != nil && len(req.TLS.PeerCertificates) > 0 {
+			req.Header.Set(xForwardedTLSClientCert, getXForwardedTLSClientCert(ctx, req.TLS.PeerCertificates))
+		} else {
+			logger.Warn("Tried to extract a certificate on a request without mutual TLS")
+		}
+	}
+
+	if p.info != nil {
+		if req.TLS != nil && len(req.TLS.PeerCertificates) > 0 {
+			headerContent := p.getXForwardedTLSClientCertInfo(ctx, req.TLS.PeerCertificates)
+			req.Header.Set(xForwardedTLSClientCertInfo, url.QueryEscape(headerContent))
+		} else {
+			logger.Warn("Tried to extract a certificate on a request without mutual TLS")
+		}
+	}
 
-	p.modifyRequestHeaders(ctx, req)
 	p.next.ServeHTTP(rw, req)
 }
 
+// getXForwardedTLSClientCertInfo Build a string with the wanted client certificates information
+// like Subject="C=%s,ST=%s,L=%s,O=%s,CN=%s";NB=%d;NA=%d;SAN=%s;
```

Comment:
As discussed, maybe instead of (or in addition to) the example, we here explain the syntax, i.e. comma to separate the certificates, semi-colon to separate the fields, and comma again to separate the fields within a field value.